src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.WorkspaceHost.cs

```diff
@@ -31,10 +31,13 @@ private class WorkspaceHost : ForegroundThreadAffinitizedObject, IVisualStudioWo
                 _workspace = workspace;
                 _client = client;
                 _currentSolutionId = workspace.CurrentSolution.Id;
+            }
 
+            public Task InitializeAsync()
+            {
                 // Ensure that we populate the remote service with the initial state of
                 // the workspace's solution.
-                RegisterPrimarySolutionAsync().Wait();
+                return RegisterPrimarySolutionAsync();
```

Comment:
What prevents this from running at the same time as OnSolutionAdded or OnAfterWorkingFolderChange below?